cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -501,16 +501,27 @@ def set_final_status():
             msg = 'Stopped services: {}'.format(','.join(failing_services))
             hookenv.status_set('blocked', msg)
             return
+    else:
+        # if we don't have components starting, we're waiting for that and
+        # shouldn't fall through to Kubernetes master running.
+        if (is_state('cni.available')):
+            hookenv.status_set('blocked',
+                               'Waiting for master components to start')
+        else:
+            hookenv.status_set('blocked',
+                               'Waiting for CNI plugins to become available')
+        return
```

Comment:
@hyperbolic2346, sorry i didn't note this before, but we're not *blocked* in this else block:

https://docs.jujucharms.com/2.4/en/reference-status

The first `if cni.available` should be `status_set('maintenance', 'waiting for master ...')` to signify that the master charm is still doing stuff, but the components haven't started yet.

The `else` should be `status_set('waiting', 'waiting for cni ...')` to signify that the master charm isn't doing anything except waiting for the cni charm to become available.

In both cases, no user intervention is required, hence we're not `blocked`.